### PR TITLE
Add worker services to the userguide, samples and release notes

### PR DESCRIPTION
**Pull request by ghale**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -141,6 +141,35 @@ In the same vein, doing file system operations such as `copy()`, `sync()` and `d
...
 See the [user manual](userguide/custom_gradle_types.html#service_injection) for how to inject services and the [`FileSystemOperations`](javadoc/org/gradle/api/file/FileSystemOperations.html) and [`ExecOperations`](javadoc/org/gradle/process/ExecOperations.html) api documentation for more details and examples.
 
+### Services available to Worker API actions
+
+The following services are now available for injection in `WorkAction` classes:
+- [ObjectFactory](javadoc/org/gradle/api/model/ObjectFactory.html)
+- [FileSystemOperations](javadoc/org/gradle/api/file/FileSystemOperations.html)
+- [ExecOperations](javadoc/org/gradle/process/Execoperations.html)
+
+These services can be injected by adding them to the constructor arguments of the `WorkAction` implementation:
```
Comment (lacasseio) [inline subprojects/docs/src/docs/release/notes.md]: ❓ Does injecting through methods also works?
```
@Inject
protected FileSystemOperations getFileSystemOperations() {
    throw new UnsupportedOperationException()
}
```

File: subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/AbstractWorkerExecutorSampleIntegrationTest.groovy [hunk 2]
```diff
@@ -41,6 +41,7 @@ abstract class AbstractWorkerExecutorSampleIntegrationTest extends AbstractInteg
     def "creates expected content (#dsl)"() {
         when:
         executer.inDirectory(workerExecutorSample(dsl))
+        executer.withArgument("--stacktrace")
```
Comment (lacasseio) [inline subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/AbstractWorkerExecutorSampleIntegrationTest.groovy]: ❓ Is this required or just a leftover from debugging an issue?

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (ghale) [reply subprojects/docs/src/docs/release/notes.md]: It does and it is discussed in the user guide manual section on "Service Injection" but the constructor injection is generally what we would recommend to users.  We would recommend the getter injection in situations where a high degree of inheritance created some awkwardness in passing constructor arguments up the chain.  This doesn't seem likely to be the case in most `WorkAction` implementations I would think.  I think I'd prefer to advertise the constructor injection in the release notes and just leave mention of getter injection in the service injection section of the user guide.

File: subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/AbstractWorkerExecutorSampleIntegrationTest.groovy (hunk 2 above)
Comment (ghale) [reply subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/AbstractWorkerExecutorSampleIntegrationTest.groovy]: Yeah, it was just leftover - I've removed it.

File: subprojects/docs/src/samples/workerApi/noIsolation/groovy/build.gradle [hunk 3]
```diff
@@ -19,14 +19,25 @@ import javax.inject.Inject
 // The parameters for a single unit of work
 interface ReverseParameters extends WorkParameters {
     Property<File> getFileToReverse()
-    Property<File> getDestinationFile()
+    Property<File> getDestinationDir()
```
Comment (big-guy) [inline subprojects/docs/src/samples/workerApi/noIsolation/groovy/build.gradle]: Can we fix the samples to use `RegularFileProperty`/`DirectoryProperty` vs `Provider<File>`?

Comment (ghale) [reply subprojects/docs/src/samples/workerApi/noIsolation/groovy/build.gradle]: Yep, done.

**Pull request merged**